pkgs/development/compilers/circt/circt-llvm.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv
+, cmake
+, python3
+, git
+, ninja
+, circt
+, llvm
+}:
+
+let
+  pythonEnv = python3.withPackages (ps: [ ps.psutil ]);
+in
+stdenv.mkDerivation {
+  pname = circt.pname + "-llvm";
+  version = circt.version;
+  src = circt.src + /llvm;
+
+  requiredSystemFeatures = [ "big-parallel" ];
+
+  nativeBuildInputs = [ cmake ninja git pythonEnv ];
+
+  cmakeDir = "../llvm";
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=ON"
+    "-DLLVM_ENABLE_BINDINGS=OFF"
+    "-DLLVM_ENABLE_OCAMLDOC=OFF"
+    "-DLLVM_BUILD_EXAMPLES=OFF"
+    "-DLLVM_OPTIMIZED_TABLEGEN=ON"
+    "-DLLVM_ENABLE_PROJECTS=mlir"
+    "-DLLVM_TARGETS_TO_BUILD="
+    "-DLLVM_INSTALL_UTILS=ON"
+  ];
+
+  outputs = [ "out" "lib" "dev" ];
+
+  postInstall = ''
+    moveToOutput "bin/llvm-config*" "$dev"
+    moveToOutput "lib" "$lib"
+    moveToOutput "lib/cmake" "$dev"
+    substituteInPlace "$dev/lib/cmake/llvm/LLVMConfig.cmake" \
+      --replace 'set(LLVM_BINARY_DIR "''${LLVM_INSTALL_PREFIX}")' 'set(LLVM_BINARY_DIR "'"$lib"'")'
+    substituteInPlace \
+      "$dev/lib/cmake/llvm/LLVMExports-release.cmake" \
+      "$dev/lib/cmake/mlir/MLIRTargets-release.cmake" \
+      --replace "$dev" "$lib"
```

Comment:
These substitutes seem not matching anything, as the build log shows

Reply:
thanks! I missed that.